### DISCUSSION TITLE
How to enable Bootstrap Grid classes (not CSS Grid based classes)?

### DISCUSSION THREAD

**Discussion by mcanouil** [quarto-dev/quarto-cli#2841] [Q&A] [answered]
Discovered in https://github.com/quarto-dev/quarto-cli/discussions/2818#discussioncomment-3872491, apparently @dragonstyle (and Quarto team) disabled by default Boostrap grid system
[src/resources/formats/html/bootstrap/_bootstrap-variables.scss#L160-L162](https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/formats/html/bootstrap/_bootstrap-variables.scss#L160-L162) in f3e33ada72bebf0ded6922f2d3f58fe4b7df1d36.

It does not seem we can enable it back.
I have tried with the following `test.scss` file:

```scss
/*-- scss:defaults --*/
$enable-grid-classes: true;
```

And the following Quarto document:

````md
---
title: "Untitled"
format: html
theme: 
  - test.scss
  - cosmo
---

```{=html}
<div class="container">
  <div class="row">
    <div class="col">
      Column
    </div>
    <div class="col">
      Column
    </div>
    <div class="col">
      Column
    </div>
  </div>
</div>
```
````

It seems I am missing something but I can't figure out what right now ...

**Comment by dragonstyle** [comment, marked as answer]
Part of the internals for Quarto layout use bootstrap layout classes using the grid system (which supports `g-` prefixed classes for performing layout), so we depend upon grid being enabled in bootstrap (and thus do not respect your variable declaration, instead always enabling and disabling the flex based css classes).

In the bootstrap grid system, you can still use classes as described here: https://getbootstrap.com/docs/5.1/layout/css-grid/#examples

**Reply by mcanouil** [reply]
> Heads up—our CSS Grid system is experimental and opt-in as of v5.1.0!

I get that but the CSS Grid is experimental even if it's using more modern CSS tools.
Quarto defines some other non-defaut Boostrap behaviour using SCSS/SASS which I think should be at least described to let users know that they won't be able to do some things usually possible in default Boostrap.

Also the version stated in <https://quarto.org/license.html> is not correct.
Internally, Quarto is using Boostrap 5.1.3. (See quarto-dev/quarto-web#445)

**Reply by dragonstyle** [reply]
Agree that we should document this more clearly! I suspect Bootstrap will be slow to migrate to grid since there are currently incompatible behaviors (e.g. gap and so on) that will be difficult to reconcile in a backward compatible way IMO. TY for PR regarding license!

**Reply by mcanouil** [reply]
I am marking this as the answer.
